rust/agents/relayer/src/msg/op_submitter.rs

```diff
@@ -186,9 +186,21 @@ async fn receive_task(
         // make sure things are getting wired up correctly; if this works in testing it
         // should also be valid in production.
         debug_assert_eq!(*op.destination_domain(), domain);
-        prepare_queue
-            .push(op, Some(PendingOperationStatus::FirstPrepareAttempt))
-            .await;
+        let status = match op.origin_db().retrieve_status_by_message_id(&op.id()) {
+            Ok(status) => status,
+            Err(e) => {
+                warn!(error=?e, "Failed to retrieve status for message");
+                None
+            }
+        };
+        let status = status.unwrap_or_else(|| {
+            debug!(
```

Comment:
nit: iiuc we'd print this for each new message, wondering if we should make this trace? just seeing as this isn't super important to know when debugging